### PR TITLE
Support reporting tool v2 to send billing data to odoo16

### PR DESCRIPTION
**Pull request by HappyTetrahedron**
Supersedes https://github.com/appuio/component-appuio-cloud-reporting/pull/86

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -10,106 +10,46 @@ parameters:
...
+      metered_billing_endpoint: null
+      oauth:
+        token_endpoint: null
+        client_id: null
+        client_secret: null
 
-    tenantmapping:
-      dry_run: true
-      metrics_selector: namespace="appuio-control-api-production"
+    rules: {}
```
Comment (simu) [inline class/defaults.yml:35]: Do I understand correctly that we'll have to manage all the billing rules in the global defaults?

**Review by simu — APPROVED** [review]
Overall change LGTM. Might make sense to have a 2nd review from someone who reviewed the Go implementation

File: class/defaults.yml (hunk 1 above)
Comment (simu) [reply class/defaults.yml:35]: Generally, I'm in favor of anything that reduces the amount of large YAML configurations that we need to manage somewhere

Comment (HappyTetrahedron) [reply class/defaults.yml:35]: well it'd be either in the component or in the hierarchy, but we'll have to manage it regardless...

**Comment by HappyTetrahedron** [comment]
* Queries are now configured in the component defaults, similar to how alerts work
* Tests aren't part of this yet, I decided to create a follow-up for re-implementing our query tests in the component

**Pull request merged**